internal/librariangen/README.md

```diff
@@ -39,6 +39,107 @@ librariangen generate \
 
 4.  **Output:** All generated files (`*.pb.go`, `*_gapic.go`, etc.) are written to the `/output` directory. The Librarian tool is then responsible for copying these files to their final destination in the `google-cloud-go` repository.
 
+## Running
+
+There are three primary ways to run the generator, with varying levels of setup complexity.
+
+### Run Librarian with the prebuilt container image
+
+This is the standard way to run the generator, using a pre-built image from Google's Artifact Registry. This method orchestrates the code generation using the `librarian` CLI, which pulls and runs the container for you.
+
+1.  **Prerequisites:**
+    *   You must have `docker` and `git` installed.
+    *   Set the `GOOGLE_CLOUD_GO_DIR` environment variable to the absolute path of your `google-cloud-go` repository checkout.
+    *   Set the `GOOGLEAPIS_DIR` environment variable to the absolute path of your `googleapis` repository checkout.
+
+2.  **Prepare the Target Repository:**
+    Before running the generator, ensure your target `google-cloud-go` repository is clean. It must also contain a committed `.librarian/state.yaml`.
+    ```bash
+    pushd $GOOGLE_CLOUD_GO_DIR && git reset --hard HEAD && git clean -fd && popd
+    ```
+
+3.  **Execute:**
+    Run the `librarian` command from the `internal/librariangen` directory. This command will generate the `secretmanager` client library using the public container image.
+    ```bash
+    go run ./cmd/librarian generate \
+      --image="gcr.io/cloud-devrel-public-resources/librarian-go:infrastructure-public-image-latest" \
+      --repo="$GOOGLE_CLOUD_GO_DIR" \
+      --library=secretmanager \
+      --api=google/cloud/secretmanager/v1,google/cloud/secretmanager/v1beta2 \
+      --api-source="$GOOGLEAPIS_DIR"
+    ```
+
+
+### Build the container yourself, and run Librarian with your image
+
+If you have made local changes to `librariangen` and want to test them in a containerized environment, you can build the Docker image locally and run it directly.
+
+1.  **Prerequisites:**
+    *   You must have `docker` and `git` installed.
+    *   Set the `GOOGLEAPIS_DIR` environment variable to the absolute path of your `googleapis` repository checkout.
+    *   You may need to authenticate with Google Artifact Registry to pull the base image: `gcloud auth configure-docker`.
```

Comment:
This did not work for you yesterday, I would include the program that did.

Reply:
Fixed, thanks!